connectors/airwatch/src/main/java/com/vmware/connectors/airwatch/AirWatchController.java

```diff
@@ -115,8 +118,18 @@ public AirWatchController(AsyncRestOperations rest, CardTextAccessor cardTextAcc
 
         String udid = cardRequest.getTokenSingleValue(UDID_KEY);
         String clientPlatform = cardRequest.getTokenSingleValue(PLATFORM_KEY);
+
+        if (StringUtils.isAnyBlank(udid, clientPlatform)) {
+            logger.debug("Either device UDID or client platform is blank.");
+            return Single.just(new ResponseEntity<>(BAD_REQUEST));
+        }
+
         Set<String> appKeywords = cardRequest.getTokens("app_keywords");
 
+        if (CollectionUtils.isEmpty(appKeywords)) {
```

Comment:
Is this really necessary? Looks like the code below ought to be able to handle empty `appKeywords` just fine.

Reply:
Removed `CollectionUtils` and added a check to make sure this token exists in the request.